core-js/src/main/javascript/cdf/queries/SolrQuery.js

```diff
@@ -116,179 +118,148 @@ define([
     get defaults() {
       return this._defaults;
     },
-    // endregion
 
-    // region Public Methods
-    /** @Override */
-    init: function(options) {
-      if (options.endpoint == null || options.collection == null) {
-        throw 'InvalidQuery';
-      }
-
-      Object.keys(options)
-        .filter(function(name) {
-          return Utils.normalizeValue(options[name]) !== null;
-        })
-        .forEach(function(name) {
-          this.setOption(name, options[name]);
-        }, this);
+    /**
+     * Gets the object containing the key/value pair
+     * of the parameters macros used in Solr Query Request.
+     *
+     * @return {object} The macros' values in the request.
+     */
+    getParameterMacros: function() {
+      var parameterMacros = {};
 
-      this.setOption("url", this.__buildSolrUrl());
-    },
+      var macros = this.getOption("macros");
```

Comment:
I don't understand where this `macros` array comes from. Shouldn't this be the result of analysis of the several specified data source property values?